qiskit/pulse/schedule.py

```diff
@@ -46,19 +47,31 @@ class Schedule(ScheduleComponent):
     prefix = 'sched'
 
     def __init__(self, *schedules: List[Union[ScheduleComponent, Tuple[int, ScheduleComponent]]],
-                 name: Optional[str] = None):
+                 name: Optional[str] = None,
+                 inplace: bool = None):
         """Create an empty schedule.
 
         Args:
             *schedules: Child Schedules of this parent Schedule. May either be passed as
                         the list of schedules, or a list of ``(start_time, schedule)`` pairs.
             name: Name of this schedule. Defaults to an autogenerated string if not provided.
+            inplace: If True, operations modify this schedule, if False, operations leave this
+                     Schedule unchanged and return a new Schedule.
         """
         if name is None:
             name = self.prefix + str(next(self.instances_counter))
             if sys.platform != "win32" and not is_main_process():
                 name += '-{}'.format(mp.current_process().pid)
 
+        self.inplace = inplace
+        if inplace is None:
+            self.inplace = False
+            warnings.warn("Schedules have been immutable by default. For performance, Schedules "
```

Comment:
Could we have this warning set a class flag so that this is raised only once per-Python interpreter session?